docs/add-new-api.md

```diff
@@ -0,0 +1,111 @@
+# How to add a new API
+
+It might happen that a new API in Elasticsearch is not yet defined
+in this repository, or we do have an endpoint definition in [`/specification/specs/_json_spec`](../specification/specs/_json_spec)
+but we don't have a type definition for it.
+In this document you will see how to add a new endpopint and how to add a new endpoint definition.
+
+## How to add a new endpoint
+
+Add a new endpoint is straightforward, you only need to copy-paste the json rest-api-spec defintion
+from the Elasticsearch repository inside [`/specification/specs/_json_spec`](../specification/specs/_json_spec)
+and you are good to go.
+
+You can find the rest-api-spec definitions [here](https://github.com/elastic/elasticsearch/tree/master/rest-api-spec/src/main/resources/rest-api-spec/api)
```

Comment:
Link to `7.x` instead of master? Also add a note that we are currently tracking `7.x` only at the top?